packages/token-bridge-sdk/src/hooks/useBalance.ts

```diff
@@ -0,0 +1,41 @@
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { useBalanceContext } from '../context/balanceContext'
+import { BigNumber } from 'ethers'
+import useSWR from 'swr'
+
+const useBalance = ({
+  provider,
+  walletAddress
+}: {
+  provider: JsonRpcProvider
+  walletAddress: string
+}) => {
+  const [allBalances, setBalances] = useBalanceContext()
+  const balances = allBalances[walletAddress]?.[provider.network?.chainId] || {
```

Comment:
I don't think we need the optional chaining operator for `provider.network`, it should always be there?

Reply:
We shouldn't be defaulting to 0, we need to differentiate between actual zero balances and data not yet being there. We can default to `null`.

Reply:
I think there were cases when `network` was undefined and it was throwing an error. I'll check if that's still the case.

Reply:
When I refresh it does throw an error (it works fine with hot reload).